docs/website/docs/dlt-ecosystem/verified-sources/pg_replication.md

```diff
@@ -0,0 +1,253 @@
+   ```
+    
+2. Credentials can be set as shown above. Alternatively, you can provide credentials in the `secrets.toml` file as follows:
+    
+   ```toml
+   sources.pg_replication.credentials="drivername://username@password.host:port/database"
+   ```
+
+3. Finally, follow the instructions in [Destinations](https://dlthub.com/docs/dlt-ecosystem/destinations/) to add credentials for your chosen destination. This will ensure that your data is properly routed.
+
+For more information, read the [General Usage: Credentials.](https://dlthub.com/docs/general-usage/credentials)
+
+## Run the pipeline
+
+1. Before running the pipeline, ensure that you have installed all the necessary dependencies by running the command:
+   ```sh
+   pip install -r requirements.txt
+   ```
+2. You're now ready to run the pipeline! To get started, run the following command:
+   ```sh
+   python pg_replication_pipeline.py
+   ```
+3. Once the pipeline has finished running, you can verify that everything loaded correctly by using the following command:
+   ```sh
+   dlt pipeline <pipeline_name> show
+   ```
+   For example, the `pipeline_name` for the above pipeline example is `pg_replication_pipeline`, you may also use any custom name instead.
+
+
+   For more information, read the guide on [how to run a pipeline](https://dlthub.com/docs/walkthroughs/run-a-pipeline).
+    
+
+## Sources and resources
+
+`dlt` works on the principle of [sources](https://dlthub.com/docs/general-usage/source) and [resources](https://dlthub.com/docs/general-usage/resource).
+
+### Resource `replication_resource`
+
+This resource yields data items for changes in one or more Postgres tables.
+
+```py
+@dlt.resource(
+    name=lambda args: args["slot_name"] + "_" + args["pub_name"],
+    standalone=True,
+)
+def replication_resource(
+    slot_name: str,
+    pub_name: str,
+    credentials: ConnectionStringCredentials = dlt.secrets.value,
+    include_columns: Optional[Dict[str, Sequence[str]]] = None,
+    columns: Optional[Dict[str, TTableSchemaColumns]] = None,
+    target_batch_size: int = 1000,
+    flush_slot: bool = True,
+) -> Iterable[Union[TDataItem, DataItemWithMeta]]:
+    ...
+```
+
+`slot_name`: Replication slot name to consume messages.
+`pub_name`: Publication slot name to publish messages.
+
+`include_columns`: Maps table name(s) to sequence of names of columns to include in the generated data items. Any column not in the sequence is excluded. If not provided, all columns are included
+
+`columns`:  Maps table name(s) to column hints to apply on the replicated table(s)
+
+`target_batch_size`: Desired number of data items yielded in a batch. Can be used to limit the data items in memory.
+
+`flush_slot`:  Whether processed messages are discarded from the replication slot. The recommended value is "True".
+
+## Customization
+
+If you wish to create your own pipelines, you can leverage source and resource methods from this verified source.
+
+1. Define the source pipeline as:
```

Comment:
It's good to mention here that the source pipeline is probably only relevant when trying out the source. For production use cases, there will likely be another process mutating the Postgres database, and the user only needs to define a destination pipeline.

Reply:
added note